### DISCUSSION TITLE
pip install for only converting text to speech?

### DISCUSSION THREAD

**Discussion by ghost** [coqui-ai/TTS#4032] [General Q&A] [answered]
I'm on Debian and tried pip install TTS. About 80 packages installed fine. After that I see that it wants to install a lot of large nvidia packages. Do I need those? I'm trying to only convert text to speech, no creating or training voices. Is there a way to install only what I need for my purpose?
thanks

**Comment by eginhard** [comment]
The Nvidia/CUDA packages are also needed for inference on a GPU, which is much faster than on CPU, so you might still want to install them. Otherwise you could also install just the CPU-version of Pytorch with their instructions: https://pytorch.org/get-started/locally/

I'd also suggest to install [our fork](https://github.com/idiap/coqui-ai-TTS) instead (with `pip install coqui-tts`), which is still maintained unlike this repo. The number of mandatory dependencies to install has also been reduced there (but it still installs the CUDA packages).

**Reply by ghost** [reply]
ok. I'll try the other fork. Is there a list of the Nvidia / cuda dependencies that I can look at first?

**Reply by eginhard** [reply, marked as answer]
They are all transitive dependencies that come through Pytorch, so there is no list in Coqui, but Pytorch 2.4.1 has the following dependencies for me:
```
[[package]]
name = "torch"
version = "2.4.1"
source = { registry = "https://pypi.org/simple" }
dependencies = [
    { name = "filelock" },
    { name = "fsspec" },
    { name = "jinja2" },
    { name = "networkx" },
    { name = "nvidia-cublas-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-cuda-cupti-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-cuda-nvrtc-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-cuda-runtime-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-cudnn-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-cufft-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-curand-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-cusolver-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-cusparse-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-nccl-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "nvidia-nvtx-cu12", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "sympy" },
    { name = "triton", marker = "platform_machine == 'x86_64' and platform_system == 'Linux'" },
    { name = "typing-extensions" },
]
```
